client/server/staking.go

```diff
@@ -625,6 +627,115 @@ func (s *Server) GetValidatorsByDelegatorAddressValidatorAddress(r *http.Request
 	return queryResp, nil
 }
 
+func (s *Server) GetStakedTokenByDelegatorAddress(req *getStakedTokenByDelegatorAddressRequest, r *http.Request) (resp any, err error) {
+	queryContext, err := s.createQueryContextByHeader(r)
+	if err != nil {
+		return nil, err
+	}
+
+	bech32AccAddress, err := utils.EvmAddressToBech32AccAddress(mux.Vars(r)["delegator_address"])
+	if err != nil {
+		return nil, err
+	}
+
+	queryResp, err := keeper.NewQuerier(s.store.GetStakingKeeper()).DelegatorDelegations(queryContext, &stakingtypes.QueryDelegatorDelegationsRequest{
+		DelegatorAddr: bech32AccAddress.String(),
+		Pagination: &query.PageRequest{
+			Key:        []byte(req.Pagination.Key),
+			Offset:     req.Pagination.Offset,
+			Limit:      req.Pagination.Limit,
+			CountTotal: req.Pagination.CountTotal,
+			Reverse:    req.Pagination.Reverse,
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	var delStakedToken []DelegationStakedToken
+	for _, delResp := range queryResp.DelegationResponses {
+		valAddr, err := sdk.ValAddressFromBech32(delResp.Delegation.ValidatorAddress)
+		if err != nil {
+			return nil, err
+		}
+
+		val, err := keeper.NewQuerier(s.store.GetStakingKeeper()).GetValidator(queryContext, valAddr)
+		if err != nil {
+			return nil, err
+		}
+
+		evmOperatorAddress, err := utils.Bech32ValidatorAddressToEvmAddress(val.OperatorAddress)
+		if err != nil {
+			return nil, err
+		}
+
+		stakedToken := val.TokensFromShares(delResp.Delegation.Shares)
+		delStakedToken = append(delStakedToken, DelegationStakedToken{
+			ValidatorOperatorAddress: evmOperatorAddress,
+			StakedToken:              stakedToken,
+		})
+	}
+
+	return QueryStakedTokenByDelegatorAddressResponse{
+		DelegationStakedToken: delStakedToken,
+		Pagination:            queryResp.Pagination,
+	}, nil
+}
+
+func (s *Server) GetTotalStakedTokenByDelegatorAddress(r *http.Request) (resp any, err error) {
```

Comment:
Noticed while testing that `GetTotalStakedTokenByDelegatorAddress` and `GetStakedTokenByDelegatorAddress` when called right after staking - do not recognize the newly staked immediately. 

Only after a few seconds, does the API gives the correct staked amount. Is this expected?

Brainstorming: Perhaps when querying past blocks the code could revert if not enough time or blocks have passed since the block being queried?

Reply:
Yes, it is expected. The CL block processing the staking related events from EL is far behind 1 block of EL, so should wait for about 2-3 seconds.

> Perhaps when querying past blocks the code could revert if not enough time or blocks have passed since the block being queried?

If there is no data to return, it will return an error. Even when the data at a specific height block (due to far behind of the latest block) is pruned.